benches/benchmark/bin/src/main.rs

```diff
@@ -0,0 +1,41 @@
+#![doc = include_str!("../README.md")]
+#![allow(
+    unused_crate_dependencies,
+    reason = "this crate imports many potentially unused dependencies"
+)]
+
+mod log;
+mod print;
+mod run;
+mod timings;
+
+use cfg_if::cfg_if;
+
+/// What `main()` does:
+/// 1. Run all enabled benchmarks
+/// 2. Record benchmark timings
+/// 3. Print timing data
+///
+/// To add a new benchmark to be ran here:
+/// 1. Copy + paste a `cfg_if` block
+/// 2. Change it to your benchmark's feature flag
+/// 3. Change it to your benchmark's type
+fn main() {
+    log::init_logger();
+
+    let mut timings = timings::Timings::new();
+
+    cfg_if! {
+        if #[cfg(not(any(feature = "example")))] {
+            compile_error!("No feature specified. Use `--features $BENCHMARK_FEATURE` when building.");
+        }
+    }
+
+    cfg_if! {
+        if #[cfg(feature = "example")] {
+            run::run_benchmark::<cuprate_benchmark_example::Example>(&mut timings);
+        }
+    }
+
+    print::print_timings(&timings);
+}
```

Comment:
...are ran in `cuprate_benchmark`, which currently simply:
1. Sets up inputs
2. Starts a timer
3. Runs benchmark

for each benchmark, prints some data. What this looks like right now:

```bash
cargo r -rp cuprate-benchmark --features example
```
```bash

    Finished `release` profile [optimized] target(s) in 0.09s
     Running `target/release/cuprate-benchmark

2024-10-10T00:01:07.701083Z  INFO cuprate_benchmark::log: Log level: INFO
2024-10-10T00:01:08.830698Z  INFO run_benchmark: cuprate_benchmark::run: cuprate_benchmark_example::Example ... 0.12949288

Finished all benchmarks, printing results:

| Benchmark                          | Time (seconds) |
|------------------------------------|----------------|
| cuprate_benchmark_example::Example | 0.12949288     |
```